### PR TITLE
Check oneview availability

### LINKED ISSUE #258: Credentials on redfish.conf needed even on session authentication mode
The application at startup is creating a OneViewClient using credentials from "redfish.conf" to authenticate on OneView even on "session" authentication mode.
There is no need to create OV clients during startup, just test if they are functional (there is an unauthenticated API endpoint for that.

 GET /controller-state.json

Response:
HTTP/1.1 200 OK
Content-Type: application/json

{
  "usingAddr" : "xxx.xxx.xxx.xxx",
  "state" : "OK"
}

### PR DESCRIPTION
**Pull request by victorhugorodrigues**
Closes #258 
Depends on #281

### CODE REVIEW

File: oneview_redfish_toolkit/tests/test_util.py [hunk 1]
```diff
@@ -188,8 +188,9 @@ def test_load_registries_valid_registry_dir_valid_dict(self):
         except Exception as e:
             self.fail('Failed to load registries files: {}'.format(e.msg))
 
+    @mock.patch.object(util, 'check_oneview_availability')
     @mock.patch.object(util, 'OneViewClient')
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/tests/test_util.py]: Do we still need this OneViewClient mock?

File: oneview_redfish_toolkit/tests/test_util.py [hunk 2]
```diff
@@ -254,9 +255,13 @@ def test_get_ip(self):
             self.fail("Failed to get a valid IP Address")
 
     @mock.patch.object(util, 'OneViewClient')
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/tests/test_util.py]: Is this OneViewClient mock still required?

File: oneview_redfish_toolkit/tests/test_util.py [hunk 3]
```diff
@@ -265,20 +270,26 @@ def test_create_certs(self, oneview_client_mockup):
         self.assertTrue(os.path.exists(os.path.join("certs", "test" + ".key")))
 
     @mock.patch.object(util, 'OneViewClient')
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/tests/test_util.py]: I believe some of these OneView mocks can be removed.

File: oneview_redfish_toolkit/tests/test_util.py [hunk 4]
```diff
@@ -265,20 +270,26 @@ def test_create_certs(self, oneview_client_mockup):
...
-    def test_load_event_service_invalid_info(self, oneview_client_mockup):
+    def test_load_event_service_invalid_info(
+        self, check_ov_availability, oneview_client_mockup):
+        check_ov_availability.return_value = 0
+
         self.assertRaises(
             OneViewRedfishError, util.load_config(self.config_file))
 
+    @mock.patch.object(util, 'check_oneview_availability')
     @mock.patch.object(util, 'OneViewClient')
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/tests/test_util.py]: Check if we still need this OneViewClient mock.

File: oneview_redfish_toolkit/tests/test_util.py [hunk 5]
```diff
@@ -295,14 +306,16 @@ def test_submit_event_with_subscriber(
 
         self.assertTrue(start_mock.call_count == 2)
 
+    @mock.patch.object(util, 'check_oneview_availability')
     @mock.patch.object(util, 'OneViewClient')
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/tests/test_util.py]: Check if we still need this OneViewClient mock.

File: oneview_redfish_toolkit/util.py [hunk 6]
```diff
@@ -436,3 +441,43 @@ def dispatch_event(event):
...
+
+
+def check_oneview_availability(ov_config):
+    """Check OneView availability by doing a GET request to OneView"""
+    attempts = 3
+
+    for attempt_counter in range(attempts):
+        try:
+            connection = HTTPSConnection(
+                ov_config['ip'], context=ssl._create_unverified_context())
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/util.py]: Create a config option to allow the user to configure whether the toolkit will check invalid certificates or just ignored them.

File: oneview_redfish_toolkit/util.py [hunk 7]
```diff
@@ -436,3 +441,43 @@ def dispatch_event(event):
...
+
+            connection.request(
+                method='GET', url='/controller-state.json',
+                headers={'Content-Type': 'application/json'})
+
+            response = connection.getresponse()
+
+            if response.status != 200:
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/util.py]: The Flask API has a file with some status code constants (`from flask_api import status`).

File: oneview_redfish_toolkit/util.py [hunk 8]
```diff
@@ -436,3 +441,43 @@ def dispatch_event(event):
             globals()['delivery_retry_interval'])
 
         dispatcher.start()
+
+
+def check_oneview_availability(ov_config):
+    """Check OneView availability by doing a GET request to OneView"""
+    attempts = 3
+
+    for attempt_counter in range(attempts):
```
Comment (lhsvobodaj) [inline oneview_redfish_toolkit/util.py]: Isn't interesting to have a wait interval between each attempt?

File: oneview_redfish_toolkit/tests/test_util.py (hunk 1 above)
Comment (victorhugorodrigues) [reply oneview_redfish_toolkit/tests/test_util.py]: No, once load_config() does not instantiate it anymore. Removing it.

File: oneview_redfish_toolkit/tests/test_util.py (hunk 2 above)
Comment (victorhugorodrigues) [reply oneview_redfish_toolkit/tests/test_util.py]: No, once load_config() does not instantiate it anymore. Removing it.

File: oneview_redfish_toolkit/tests/test_util.py (hunk 3 above)
Comment (victorhugorodrigues) [reply oneview_redfish_toolkit/tests/test_util.py]: Yes. Removing it.

File: oneview_redfish_toolkit/tests/test_util.py (hunk 4 above)
Comment (victorhugorodrigues) [reply oneview_redfish_toolkit/tests/test_util.py]: We don't need it, once load_config() does not instantiate it anymore. Removing it.

File: oneview_redfish_toolkit/tests/test_util.py (hunk 5 above)
Comment (victorhugorodrigues) [reply oneview_redfish_toolkit/tests/test_util.py]: We don't need it, once load_config() does not instantiate it anymore. Removing it.

**Comment by lhsvobodaj** [comment]
Abandoned in favor of #287.

**Pull request closed without merging**